### PR TITLE
Feature/validation/add jakarta validation

### PR DESCRIPTION
**Pull request by BartoliniAndBorderCollies**
#### Issue
https://trello.com/c/L595oAu5

### Description
This branch adds `jakarta validation`, so that client needs to fulfill some requirements while providing some data to the application

### Motivation
Validations make order and are recipes to have a `consistence records` `in database`

### Changes introduced by this pull request
- add `jakarta validation` and `hibernate validator` dependencies
- add `@Valid` to various controllers `PUT` and `POST` methods
- add validations to `BikeRequestDTO` class
- add validation to `Feedback` class 
- add validations to `ChargingStation` class
- add validations to `RentRequest` class
- add validations to `RentRequestDTO` class
- add validations to `User` class
- add `CustomExceptionHandler` class to handle exception and provide a message to the client with information what went wrong
- refactor of test methods in integration tests classes to fulfill new requirements

### Testing
- [x] all validations work as expected
- [x] all test methods are functional

### CODE REVIEW

File: src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+
+@ControllerAdvice
```
[the remark this answers is not in the document]
Comment (BartoliniAndBorderCollies) [reply src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java:11]: OK you mean a deletion of line? OK

File: src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+public class CustomExceptionHandler {
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<String> handleMethodArgumentNotValidException(MethodArgumentNotValidException ex) {
+        Object[] detailMessageArguments = ex.getDetailMessageArguments();
+
+        assert detailMessageArguments != null;
```
[the remark this answers is not in the document]
Comment (BartoliniAndBorderCollies) [reply src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java:19]: wow! thanks for this!

File: src/main/java/com/klodnicki/Bike/v2/model/entity/Feedback.java [hunk 3]
```diff
@@ -18,7 +19,7 @@ public class Feedback {
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     @Column(name = "id", nullable = false)
     private Long id;
-
+    @NotBlank
```
[the remark this answers is not in the document]
Comment (BartoliniAndBorderCollies) [reply src/main/java/com/klodnicki/Bike/v2/model/entity/Feedback.java:22]: I forgot! :)

**Comment by BartoliniAndBorderCollies** [comment]
thanks for your review! I made appropriate changes take a look now!
thanks again!

File: src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (JasinskiSz) [reply src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java]: Yep, exactly.

File: src/main/java/com/klodnicki/Bike/v2/model/entity/Feedback.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (JasinskiSz) [reply src/main/java/com/klodnicki/Bike/v2/model/entity/Feedback.java]: Good, that I caught it!

File: src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java [hunk 4]
```diff
@@ -16,7 +15,9 @@ public class CustomExceptionHandler {
     public ResponseEntity<String> handleMethodArgumentNotValidException(MethodArgumentNotValidException ex) {
         Object[] detailMessageArguments = ex.getDetailMessageArguments();
 
-        assert detailMessageArguments != null;
-        return new ResponseEntity<>("Something went wrong!: \n" + detailMessageArguments[1], HttpStatus.BAD_REQUEST);
+        if (detailMessageArguments != null) {
+            return new ResponseEntity<>("Something went wrong!: \n" + detailMessageArguments[1], HttpStatus.BAD_REQUEST);
+        }
+        return new ResponseEntity<>(HttpStatus.OK);
```
Comment (JasinskiSz) [inline src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java:21]: I think that regardless of anything, when you catch exception REST API shouldn't return status 200.

Comment (BartoliniAndBorderCollies) [reply src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java:21]: of course you are right! I guess I haven't woken up yet;)

File: src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java
[the remark this answers is not in the document]
Comment (BartoliniAndBorderCollies) [reply src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java:1]: I think I know what do you mean, I replace it to controllers but put it to own package

**Comment by BartoliniAndBorderCollies** [comment]
OK great! now it should be fine!

File: src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java
[the remark this answers is not in the document]
Comment (JasinskiSz) [reply src/main/java/com/klodnicki/Bike/v2/exception/CustomExceptionHandler.java:1]: Exactly what you said :)

**Review by JasinskiSz — APPROVED** [review]
Great! All things are fixed!
Solid work, @BartoliniAndBorderCollies!

**Pull request merged**